authentication/http.go

```diff
@@ -93,31 +81,20 @@ func GetGroups(ctx context.Context) ([]string, bool) {
 	return groups, ok
 }
 
-// WithTenantMiddlewares creates a single Middleware for all
-// provided tenant-middleware sets.
-func WithTenantMiddlewares(middlewareSets ...map[string]Middleware) Middleware {
-	middlewares := map[string]Middleware{}
-
-	for _, ms := range middlewareSets {
-		for t, m := range ms {
-			middlewares[t] = m
-		}
-	}
-
+// Creates a single Middleware.
+func WithTenantMiddleware(middleware Middleware) Middleware {
```

Comment:
This function does not actually use the Middleware passed to it, so I'm worried that we are missing a chunk of logic that is actually important. Note that previously we would find the right Middleware for the tenant and then call `m(next).ServeHTTP`. To work accordingly, this function should be calling `middleware(next).ServeHTTP`

Reply:
This is part of the codebase is no more required as the specified authN Middleware per tenant is now implemented under newTenant() of main package.

Reply:
If we are not using it internally any longer then we should definitely delete this function. It's been modified not to use the given middleware so it's not useful for the community either.